File: Estruturas/1023/1023.rs
use std::io::{self, BufRead};

fn main() {
  let stdin = io::stdin();
  let mut lines = stdin.lock()
    .lines();

  let mut city = 0;
  loop {
    let line = lines.next().unwrap().unwrap();
    let n: u32 = line.trim().parse().unwrap();

    if n == 0 {
      break;
    }

    if city > 0 {
      println!();
    }

    let mut total_x = 0;
    let mut total_y = 0;
    let mut consumos = vec![0; 201];

    for _ in 0..n {
      let line = lines.next().unwrap().unwrap();
      let mut inputs = line.split_whitespace();

      let x: i32 = inputs.next()
        .and_then(|a| a.parse().ok())
        .unwrap();
      let y: i32 = inputs.next()
        .and_then(|a| a.parse().ok())
        .unwrap();

      total_x += x;
      total_y += y;
      let index = (y/x) as usize;

      consumos[index] = consumos[index] + x;
    }

    city += 1;
    println!("Cidade# {}:", city);
    
    let mut output = String::with_capacity(1024);
    for i in 0..consumos.len() {
      if consumos[i] > 0 {
        output.push_str(&format!("{}-{} ", consumos[i], i));
      }
    }

    println!("{}", output.trim_end());

    let consumo_total = ((100.0 * total_y as f64) / total_x as f64).floor() / 100.0;

    println!("Consumo medio: {:.2} m3.", consumo_total);
  }
}

File: Estruturas/1023/1023.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.trim().split('\n');

lines.pop();

let T = 0;
let p = 0;

while (p < lines.length) {
    const N = parseInt(lines[p++]);

    if(T) {
        console.log('');
    }

    let consumos = Array(201);
    consumos.fill(0);
    let [totalX, totalY] = [0, 0];
    for(let i = 0; i < N; i++){
        let [X, Y] = lines[p++].split(' ').map((x) => parseInt(x));

        totalX += X;
        totalY += Y;
        consumos[parseInt(Math.floor(Y/X))] += X;
    }

    console.log(`Cidade# ${++T}:`);

    const output = consumos.reduce((prev, curr, i) => {
        if(curr > 0) {
            prev.push(`${curr}-${i}`);
        }
        return prev
    }, []).join(' ');
    console.log(output);

    const consumo_medio = Math.floor((100 * totalY) / totalX) / 100

    console.log(`Consumo medio: ${consumo_medio.toFixed(2)} m3.`);
}